estmator_project/estmator_project/views.py

```diff
@@ -17,23 +22,47 @@ def get_context_data(self, **kwargs):
         return context
 
 
-class QuoteView(TemplateView):
-    template_name = 'quote.html'
+class AboutView(TemplateView):
+    template_name = 'about.html'
 
     def get_context_data(self, **kwargs):
-        try:
-            context = super(QuoteView, self).get_context_data(**kwargs)
-            context['categories'] = Category.objects.all()
+        context = super(AboutView, self).get_context_data(**kwargs)
+        return context
 
-            options_form = QuoteOptionsForm()
-            context['options_form'] = options_form.as_ul
 
-            context['quote_client'] = self.request.GET.get('client')
-            context['quote_name'] = self.request.GET.get('name')
-        except (KeyError, ValueError):
-            return redirect('menu')
+@login_required
+def quote_view(request):
+    if request.method == 'POST':
+        categories = {c: {p: 0 for p in c.product_set.all()} for c in Category.objects.all()}
+        if 'quote' in request.POST:
+            quote = Quote.objects.get(id=request.POST['quote'])
+            quote_name = quote.name
+            for prop in quote.productproperties_set.all():
+                categories[prop.product.category][prop.product] = prop.count
```

Comment:
It feels to me like there should be an API method on the category that handles the logic here.  It also feels like this has the potential to become a very very expensive operation, what with the nested dict comprehension up there, followed by another loop over all the product properties in a quote.